### DISCUSSION TITLE
Can't seem to add a favicon

### DISCUSSION THREAD

**Discussion by liconver** [nuxt/nuxt#16727] [Questions]
Hello, so I am actually just getting started with Nuxt 3, I never used Nuxt before and one thing that is bugging me is on my dev server I can't add a favicon.

According to the documentation it should be placed in the public folder.  In nuxt.config.ts I added
 `app: {
        head: {
            link: [
                {rel: 'icon', type: "image/x-icon", href: '/favicon.ico?v2'}
            ]
        }
    }`
to the defineNuxtConfig function.  I see this link in the head of the html in my browser but I don't see any indication that vite has included the favicon file in the built bundle.  However I don't really understand exactly what the build tool is doing here so it's hard for me to track down what the issue is.

There was a discussion about how to add a favicon in Nuxt 2 but this process seems like it has changed in Nuxt 3.

**Comment by danielroe** [comment]
In addition to specifying it in the head, you should add your favicon to `~/public/favicon.ico`.

**Reply by liconver** [reply]
Thanks for the response but I did have the favicon file in the public folder in the root directory.  Now I tried to pass the link tag via the useHead function in the app.vue file but I still can't find the favicon file in the build.

**Reply by liconver** [reply]
I'm sorry the issue was I was using a favicon generator and unzipping the output into public and it was in an additional folder that for some reason wasn't too clear in vscode's explorer.  That's all.

**Comment by shehabulalam** [comment]
Just put your `favicon.ico` in `public` folder.
Then edit your `app.vue` like this
```
useHead({
  link: [
    { rel: 'icon', type: 'image/x-icon', href: '/favicon.ico' },
 ]
})
```

**Reply by adampatterson** [reply]
This assumes you're using an ico file.

In my case I had to delete `public/favicon.ico` and then use:

```
useHead({
	link: [
		{ rel: "icon", type: "image/png", href: "/img/favicon.png" },
	],
});
```

if you don't delete the original icon you end up with two, and the browser uses the first icon `<link>`.

**Comment by mhoffrog** [comment]
See my answer on [@stackoverflow - Favicon not shown after deploy on Github Pages - Nuxt3](https://stackoverflow.com/a/79404287/15763374).
